FLIR/conservator/wrappers/collection.py

```diff
@@ -160,56 +161,80 @@ def recursively_get_children(self, include_self=False, fields=None):
+            self._conservator,
+            Image,
+            Query.images,
+            fields=fields,
+            search_text=search_text,
+            collection_id=self.id,
         )
         return images
 
-    def recursively_get_images(self, fields=None):
+    def recursively_get_images(self, fields=None, search_text=""):
         """Yields all images in this and child collections recursively"""
         for collection in self.recursively_get_children(include_self=True, fields="id"):
-            yield from collection.get_images(fields)
+            yield from collection.get_images(fields, search_text)
 
-    def get_videos(self, fields=None):
+    def get_videos(self, fields=None, search_text=""):
         """Returns a query for all videos in this collection."""
         videos = PaginatedQuery(
-            self._conservator, Video, Query.videos, fields=fields, collection_id=self.id
+            self._conservator,
+            Video,
+            Query.videos,
+            fields=fields,
+            search_text=search_text,
+            collection_id=self.id,
         )
         return videos
 
-    def recursively_get_videos(self, fields=None):
+    def recursively_get_videos(self, fields=None, search_text=""):
         """Yields all videos in this and child collections recursively"""
         for collection in self.recursively_get_children(include_self=True, fields="id"):
-            yield from collection.get_videos(fields)
+            yield from collection.get_videos(fields, search_text)
 
-    def get_media(self, fields=None):
+    def get_media(self, fields=None, search_text=""):
         """
         Yields all videos, then images in this collection.
 
         :param fields: The `fields` to include in the media. All
             fields must exist on both Image and Video types.
         """
-        yield from self.get_videos(fields=fields)
-        yield from self.get_images(fields=fields)
+        yield from self.get_videos(fields=fields, search_text=search_text)
+        yield from self.get_images(fields=fields, search_text=search_text)
 
-    def recursively_get_media(self, fields=None):
+    def recursively_get_media(self, fields=None, search_text=""):
         """Yields all videos and images in this and child collections recursively"""
         for collection in self.recursively_get_children(include_self=True, fields="id"):
-            yield from collection.get_media(fields)
+            yield from collection.get_media(fields, search_text)
 
-    def get_datasets(self, fields=None):
+    def get_datasets(self, fields=None, search_text=""):
         """Returns a query for all datasets in this collection."""
         datasets = PaginatedQuery(
             self._conservator,
             Dataset,
             Query.datasets,
             fields=fields,
+            search_text=search_text,
             collection_id=self.id,
         )
         return datasets
 
+    def remove_media(self, media_id):
```

Comment:
Might as well change this to `remove_medias` that takes a list of ID to remove; and make `remove_media` call with single ID